feature/text_based.py

```diff
@@ -0,0 +1,694 @@
+        self.featurization_method = None
+        self.cost_metric = None
+        self._validate_args(selection_size)
+
+        # Set input arguments
+        self.selection_size = selection_size
+        self.seed = seed
+        self.trials = trials
+        self.verbose = verbose == 1
+
+        # Initialize class variables
+        self.matrix = None
+        self.features = None
+        self._num_rows = None
+        self._num_cols = None
+
+    def run_content_selection(self,
+                              input_df: pd.DataFrame,
+                              categories: List[int], featurization_method: TextWiser,
+                              selection_size: int,
+                              optimization_method: str = "exact",
+                              cost_metric: str = "diverse",
+                              trials: int = 10) -> List:
+
+        """Run content selection algorithm.
+
+        Parameters
+        ----------
+        selection_size
+        input_df: pd.DataFrame
+            Input data frame with categories and features of content to select from.
+
+        categories: List[int]
+            List of columns in data that contains categories/labels to be covered.
+
+        featurization_method: TextWiser
+
+        optimization_method: str, default="exact"
+            Optimization method used to perform content selection.
+            See TextBased in Selector.py for supported options definition
+
+        cost_metric: str, default = "diverse"
+
+        trials: int, default = 10
+
+        Returns
+        -------
+        List of indices in data that are selected.
+        """
+
+        # Process data
+        self.cost_metric = cost_metric
+        self.featurization_method = featurization_method
+        self.selection_size = selection_size
+        self._process_df(input_df, categories, selection_size, optimization_method, cost_metric)
+
+        # Run multi-level set covering optimization
+        if optimization_method == "exact":
+            selected = self._select_multi_level_optimization()
+        # Run KMeans selection algorithm
+        elif optimization_method == "kmeans":
+            selected = self._select_kmeans()
+        # Run greedy heuristic
+        elif optimization_method == "greedy":
+            selected = self._select_greedy()
+        # Run random selection algorithm
+        elif optimization_method == "random":
+            selected = self._select_random(trials=trials)
+        else:
+            raise NotImplementedError(f"{optimization_method} has not been implemented.")
+
+        # Sort selection
+        selected.sort()
+
+        return selected
+
+    def _process_df(self, input_df: pd.DataFrame, categories: List[int], selection_size: int,
+                    optimization_method: str, cost_metric: str) -> NoReturn:
+        """_process_df
+        We have implemented two versions of this function. In the version here, the categories parameter
```

Comment:
aren't we removing the other one? we don't need both correct?

Reply:
Can we add some explanation here what this process is doing. I am not following it